### ISSUE TITLE
Getting started to contribute page

### ISSUE THREAD

**Issue by AbdealiJK** [jupyter/jupyterlab#751]
It would be really nice to have a "Getting Started to contribute" page for people new to the repo who want to contribute to this repo.
I am currently not sure how to start, and a wiki page for this would be awesome.

It would also be really nice to have a label for "Easy" issues which any newbie to the repository can do without spending too much time.

**Comment by blink1073** [comment]
Hi @AbdealiJK, you are right, we are lacking in the contributing guidelines, we need to reorganize to match the styles of https://github.com/jupyter/notebook/blob/master/README.md and https://github.com/jupyter/notebook/blob/master/CONTRIBUTING.rst.

As for "Easy" issues, we have been tagging them with "sprint-friendly": https://github.com/jupyter/jupyterlab/issues?q=is%3Aopen+is%3Aissue+label%3Asprint-friendly

**Comment by blink1073** [comment]
#582 is related.

**Comment by AbdealiJK** [comment]
@blink1073 The sprint-friendly label looks good !

I was trying to modify a typescript file `src/csvwidget/widget.ts` by adding a console.log() in renderTable(). I've installed jupyterlab from source and run `jupyter-lab` to start the server. I opened a new terminal and ran `npm run watch` as recommended by the `jupyter-lab` command so that typescript gets auto compiled on change.

On running the `npm run watch`, it takes a long time (more than 15mins) and after a while, my computer completely hangs (i3, 4gb RAM, 4gb swap).
Is it supposed to take a long time ? Shouldn't it stop giving out messages and wait for me to trigger a change after a bit ?
Here is the log from `npm run watch` for the 10 minutes or so after which is begins lagging a little: http://pastebin.com/LRMC9KLK
